src/interface/file.rs

```diff
@@ -122,168 +127,186 @@ impl EmulatedFile {
         }?;
 
         let absolute_filename = fs::canonicalize(&path)?;
+        let rawfd = f.as_raw_fd() as i32;
 
         OPEN_FILES.insert(filename.clone());
-        let filesize = f.metadata()?.len();
+        let filesize = f.metadata()?.len() as usize; 
 
-        Ok(EmulatedFile {filename: filename, abs_filename: absolute_filename, fobj: Some(Arc::new(Mutex::new(f))), filesize: filesize as usize})
+        let mapsize = ((filesize / MAP_1GB) + 1) * MAP_1GB;
+
+        f.set_len(mapsize as u64)?;
+
+        let emfile: Vec<u8>;
+        unsafe {
+            let filemap_addr = mmap(0 as *mut c_void, mapsize, PROT_READ | PROT_WRITE, MAP_SHARED, rawfd, 0 as i64);
+            let _syncret = msync(filemap_addr, mapsize, MS_ASYNC);
```

Comment:
You should not need to msync immediately after a mapping. The memory region starts out with the pages not mapped in memory and when a fault happens the backing file is read and the page is populated. 
Msync writes changes from the memory region back into memory, but as nothing's been written to that region of memory here, nothing happens